src/LinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -528,22 +528,44 @@ void UprobesUnwindingVisitor::Visit(uint64_t event_timestamp,
 
 void UprobesUnwindingVisitor::Visit(uint64_t event_timestamp,
                                     const UserSpaceFunctionEntryPerfEventData& event_data) {
-  function_call_manager_->ProcessFunctionEntry(event_data.tid, event_data.function_id,
-                                               event_timestamp, std::nullopt);
+  if (!tid_to_root_namespace_tid_.contains(event_data.tid)) {
+    ORBIT_ERROR(
+        "Recieved function entry event from unknown thread with tid %d. Dropping this event.",
+        event_data.tid);
+    return;
+  }
+  const pid_t tid = tid_to_root_namespace_tid_[event_data.tid];
+
+  function_call_manager_->ProcessFunctionEntry(tid, event_data.function_id, event_timestamp,
+                                               std::nullopt);
 
-  return_address_manager_->ProcessFunctionEntry(event_data.tid, event_data.sp,
-                                                event_data.return_address);
+  return_address_manager_->ProcessFunctionEntry(tid, event_data.sp, event_data.return_address);
 }
 
 void UprobesUnwindingVisitor::Visit(uint64_t event_timestamp,
                                     const UserSpaceFunctionExitPerfEventData& event_data) {
-  std::optional<FunctionCall> function_call = function_call_manager_->ProcessFunctionExit(
-      event_data.pid, event_data.tid, event_timestamp, std::nullopt);
+  if (!tid_to_root_namespace_tid_.contains(event_data.tid)) {
+    ORBIT_ERROR(
+        "Recieved function exit event from unknown thread with tid %d. Dropping this event.",
+        event_data.tid);
+    return;
+  }
+  if (!tid_to_root_namespace_tid_.contains(event_data.pid)) {
+    ORBIT_ERROR(
+        "Recieved function exit event from unknown process with pid %d. Dropping this event.",
+        event_data.pid);
+    return;
+  }
```

Comment:
These logs are problematic: one single thread without mapping can heavily spam the logs and completely tank the performance of OrbitService.